### PR TITLE
Updated dash packages and added pages to pa11y checks

### LINKED ISSUE #8704: Add a concept page and an item-viewer page to Pa11y
Pa11y doesn't check a concept page and an item-viewer page so we'll add that in.

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Devs/Team

## What is it doing for them?
Added a concept page and an item viewer page for pa11y checks; ended up updating packages in dash as well + fixing some type errors in all dash pages.

Once this is approved I'll test deploying the changes properly.

Closes #8704

### CODE REVIEW

File: dash/webapp/pages/pa11y.tsx [hunk 1]
```diff
@@ -35,7 +36,7 @@ const Issue = styled.div`
 `;
 
 const Index: FC = () => {
-  const [resultsList, setResultsList] = useState(null);
+  const [resultsList, setResultsList] = useState([]);
```
Comment (rcantin-w) [inline dash/webapp/pages/pa11y.tsx:39]: This was giving `Property 'results' does not exist on type 'never'.` on line 62, so I changed how we treat it a bit - open to suggestions!

File: pa11y/webapp/package.json [hunk 2]
```diff
@@ -1,9 +1,6 @@
 {
   "name": "@weco/pa11y",
   "scripts": {
-    "dev": "next",
```
Comment (rcantin-w) [inline pa11y/webapp/package.json:4]: we don't use next in the pa11y project

File: dash/webapp/next-env.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/// <reference types="next" />
+/// <reference types="next/types/global" />
+/// <reference types="next/image-types/global" />
+
+// NOTE: This file should not be edited
+// see https://nextjs.org/docs/basic-features/typescript for more information.
```
Comment (alexwlchan) [inline dash/webapp/next-env.d.ts:6]: If this file is empty do we need it at all? Or do the comments do something?

File: dash/webapp/pages/index.tsx [hunk 4]
```diff
@@ -1,116 +1,125 @@
...
+              <a
+                href="https://github.com/wellcomecollection/wellcomecollection.org"
+                rel="repo"
+              >
+                Experience Github repo
+              </a>
+            </li>
+            <li>
+              <a
+                href="https://github.com/wellcomecollection/platform"
```
Comment (alexwlchan) [inline dash/webapp/pages/index.tsx:41]: Not for today, but half these links are broken and need pruning.

**Review by alexwlchan — APPROVED** [review]
I'd suggest also uncommenting the `deploy dash` task in `.buildkite/pipeline.yml` – this will deploy changes in CI, and we'll be more likely to spot the next time this breaks.

File: dash/webapp/package.json [hunk 5]
```diff
@@ -11,7 +11,7 @@
   },
   "dependencies": {
     "@babel/core": "^7.2.0",
-    "@types/react": "^com",
+    "@types/react": "18.0.1",
```
Comment (rcantin-w) [inline dash/webapp/package.json:14]: No idea what happened with the above.

File: pa11y/webapp/write-report.js [hunk 6]
```diff
@@ -22,7 +22,6 @@ const urls = [
   'https://wellcomecollection.org/events/Wqkd1yUAAB8sW4By',
   'https://wellcomecollection.org/event-series/WlYT_SQAACcAWccj',
   'https://wellcomecollection.org/concepts/n4fvtc49',
-  'https://wellcomecollection.org/works/a2239muq/items',
```
Comment (rcantin-w) [inline pa11y/webapp/write-report.js]: We're already testing `https://wellcomecollection.org/works/e7vav3ss/items?page=1&canvas=1&sierraId=b28136615&langCode=eng`. Is that what you meant for adding an item viewer page @alexwlchan ? I might've gotten it wrong.

File: dash/webapp/next-env.d.ts (hunk 3 above)
Comment (rcantin-w) [reply dash/webapp/next-env.d.ts:6]: It was automatically created when installing packages, we have the same in `content` and `catalogue`.

File: pa11y/webapp/write-report.js (hunk 6 above)
Comment (alexwlchan) [reply pa11y/webapp/write-report.js]: Yup, that looks fine, although I have no memory of suggesting that. 😅

**Pull request merged**